src/engine/cachingreader/cachingreaderchunk.cpp

```diff
@@ -215,8 +215,8 @@ void CachingReaderChunkForOwner::removeFromList(
     }
 
     // Disconnect this chunk from the double-linked list
-    const auto pPrev = m_pPrev;
-    const auto pNext = m_pNext;
+    auto* const pPrev = m_pPrev;
+    auto* const pNext = m_pNext;
```

Comment:
You don't have to look up the types, the compiler already does that for you.

Btw, our guidelines for using of `auto` are unsound. Passing a result from a function on is ok, but storing it in a local auto variable before passing it on is not. I suggest to revise them. The compiler and the linters are much better at checking types than us human.

This repeated nitpicking about the usage of `auto` is an indicator that our rules should be revised. This adds unneeded noise to reviews and it really bugs me.

Reply:
Even AUTOSAR is more permissive: https://www.autosar.org/fileadmin/user_upload/standards/adaptive/17-10/AUTOSAR_RS_CPP14Guidelines.pdf

Reply:
I am afraid we are not on the same page here. I would prefer to use less auto you like to prefer to use more. 

Without auto, the the intended type becomes visible. If a programmer has a different opinion that the compiler we get et least a warning that is good. When you write the code you know the type anyway, so write it down, this is reasonable effort because you write the code only one time while it is read multiple times.  
When writing the coder you have the full power of an IDE. If you review it you have to do two Github searches to find out the type by a chance.

Reply:
I am not able to guess what a developer understands when you say "not in view". A member variable inside a class is supposed to be "in view" for me. Only a local variable is closer.

Without provable rules this is annoying.

Reply:
> I am unsure here. I guess the original code was meant to be:
> const CachingReaderChunkForOwner* pPrev = m_pPrev;
> and you have discovered a bug.

Yes, when I saw clang-tidy applying this fix I was worried I misconfigured it because leading `const` had been removed. Then I looked at the code below and began to wonder why it didn't throw a compiler warning...

Reply:
Writing the `const` classifier on the right side of type declarations (like it is done for functions) would prevent such misinterpretations. Unfortunately, I did not succeed to propose this style for Mixxx. We still use the classical const-to-the-left rule for data type declarations, causing many question about to which part of the type the `const` applies.

Reply:
So what should I change here? Or leave it as-is? Please add a suggestion.

Reply:
`auto* const` is correct. The pointer is const, not the referenced value.